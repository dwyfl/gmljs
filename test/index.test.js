import GML from '../src';
import gmlData from './data/example000.xml';

it('parses a minimal GML file correctly', () => {
  const gml = new GML(gmlData).toString();
  expect(gml).toMatchSnapshot();
});