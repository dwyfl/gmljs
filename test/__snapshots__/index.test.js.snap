// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`creates an empty GML document 1`] = `"<gml spec=\\"1.0\\"><tag><drawing></drawing></tag></gml>"`;

exports[`parses a basic GML document correctly 1`] = `"<gml><tag><header><client><name>seen</name></client></header><environment><offset><x>-58</x><y>173</y><z>0.96</z></offset><rotation><x>21</x><y>0</y><z>0</z></rotation></environment><drawing><stroke><pt><x>0.406848</x><y>0.303721</y><z>0</z><t>0</t></pt><pt><x>0.407443</x><y>0.30549</y><z>16</z><t>0.16</t></pt><pt><x>0.408944</x><y>0.307128</y><z>56.000103</z><t>0.560001</t></pt><pt><x>0.411309</x><y>0.309808</y><z>59.199902</z><t>0.591999</t></pt><pt><x>0.415111</x><y>0.313588</y><z>62.5</z><t>0.625</t></pt><pt><x>0.418266</x><y>0.317968</y><z>65.800095</z><t>0.658001</t></pt><pt><x>0.419495</x><y>0.321312</y><z>70.600098</z><t>0.706001</t></pt><pt><x>0.417704</x><y>0.322598</y><z>77.099998</z><t>0.771</t></pt><pt><x>0.413242</x><y>0.321135</y><z>80.300095</z><t>0.803001</t></pt><pt><x>0.40648</x><y>0.318537</y><z>83.499901</z><t>0.834999</t></pt><pt><x>0.399285</x><y>0.315566</y><z>86.700096</z><t>0.867001</t></pt><pt><x>0.392792</x><y>0.312797</y><z>89.899803</z><t>0.898998</t></pt><pt><x>0.389448</x><y>0.311699</y><z>93.199905</z><t>0.931999</t></pt><pt><x>0.389362</x><y>0.313375</y><z>96.400101</z><t>0.964001</t></pt><pt><x>0.392459</x><y>0.317153</y><z>99.499893</z><t>0.994999</t></pt><pt><x>0.398701</x><y>0.3233</y><z>102.699997</z><t>1.027</t></pt><pt><x>0.407917</x><y>0.330721</y><z>106.000099</z><t>1.060001</t></pt><pt><x>0.419253</x><y>0.338284</y><z>109.199905</z><t>1.091999</t></pt><pt><x>0.43108</x><y>0.345692</y><z>113.800003</z><t>1.138</t></pt><pt><x>0.440992</x><y>0.350054</y><z>115.300003</z><t>1.153</t></pt><pt><x>0.445643</x><y>0.348677</y><z>120.299904</z><t>1.202999</t></pt><pt><x>0.444271</x><y>0.341277</y><z>126.699806</z><t>1.266998</t></pt><pt><x>0.438577</x><y>0.330359</y><z>129.899994</z><t>1.299</t></pt><pt><x>0.429677</x><y>0.317536</y><z>133.200104</z><t>1.332001</t></pt><pt><x>0.419238</x><y>0.303873</y><z>136.299896</z><t>1.362999</t></pt><pt><x>0.408579</x><y>0.290866</y><z>139.600006</z><t>1.396</t></pt><pt><x>0.398055</x><y>0.278508</y><z>142.700195</z><t>1.427002</t></pt><pt><x>0.390571</x><y>0.268285</y><z>145.899994</z><t>1.459</t></pt><pt><x>0.388823</x><y>0.262584</y><z>149.100098</z><t>1.491001</t></pt><pt><x>0.39341</x><y>0.26178</y><z>152.299896</z><t>1.522999</t></pt><pt><x>0.403062</x><y>0.265453</y><z>155.5</z><t>1.555</t></pt><pt><x>0.416395</x><y>0.272771</y><z>158.800095</z><t>1.588001</t></pt><pt><x>0.430193</x><y>0.281506</y><z>163.700104</z><t>1.637001</t></pt><pt><x>0.441986</x><y>0.290539</y><z>165.300003</z><t>1.653</t></pt><pt><x>0.449884</x><y>0.298542</y><z>170.100006</z><t>1.701</t></pt><pt><x>0.451911</x><y>0.303411</y><z>173.300201</z><t>1.733002</t></pt><pt><x>0.44821</x><y>0.304108</y><z>176.499908</z><t>1.764999</t></pt><pt><x>0.43988</x><y>0.301209</y><z>179.700104</z><t>1.797001</t></pt><pt><x>0.426785</x><y>0.295182</y><z>182.899902</z><t>1.828999</t></pt><pt><x>0.411153</x><y>0.287147</y><z>186.199997</z><t>1.862</t></pt><pt><x>0.394617</x><y>0.279144</y><z>189.400101</z><t>1.894001</t></pt><pt><x>0.379989</x><y>0.272351</y><z>192.700195</z><t>1.927002</t></pt><pt><x>0.37017</x><y>0.267195</y><z>195.899994</z><t>1.959</t></pt><pt><x>0.368514</x><y>0.265214</y><z>199.100098</z><t>1.991001</t></pt><pt><x>0.374979</x><y>0.266611</y><z>205.400101</z><t>2.054001</t></pt><pt><x>0.386409</x><y>0.269996</y><z>210.200104</z><t>2.102001</t></pt><pt><x>0.400604</x><y>0.274261</y><z>213.399902</z><t>2.133999</t></pt><pt><x>0.415639</x><y>0.277886</y><z>216.600006</z><t>2.166</t></pt><pt><x>0.42982</x><y>0.280867</y><z>219.800217</z><t>2.198002</t></pt><pt><x>0.440918</x><y>0.282027</y><z>223</z><t>2.23</t></pt><pt><x>0.447547</x><y>0.280212</y><z>226.300003</z><t>2.263</t></pt><pt><x>0.448725</x><y>0.275588</y><z>229.600098</z><t>2.296001</t></pt><pt><x>0.443143</x><y>0.267973</y><z>232.699997</z><t>2.327</t></pt><pt><x>0.432135</x><y>0.259152</y><z>237.400101</z><t>2.374001</t></pt><pt><x>0.418085</x><y>0.250251</y><z>238.999908</z><t>2.389999</t></pt><pt><x>0.403406</x><y>0.243093</y><z>243.800003</z><t>2.438</t></pt><pt><x>0.391254</x><y>0.239535</y><z>247.199997</z><t>2.472</t></pt><pt><x>0.385113</x><y>0.239543</y><z>250.400223</z><t>2.504002</t></pt><pt><x>0.386082</x><y>0.24363</y><z>253.599899</z><t>2.535999</t></pt><pt><x>0.394325</x><y>0.251875</y><z>256.80011</z><t>2.568001</t></pt><pt><x>0.407361</x><y>0.262099</y><z>260.100189</z><t>2.601002</t></pt><pt><x>0.422346</x><y>0.272695</y><z>263.299896</z><t>2.632999</t></pt><pt><x>0.438438</x><y>0.283202</y><z>266.500092</z><t>2.665001</t></pt><pt><x>0.454149</x><y>0.29115</y><z>269.69989</z><t>2.696999</t></pt><pt><x>0.46682</x><y>0.295764</y><z>273</z><t>2.73</t></pt><pt><x>0.475956</x><y>0.297442</y><z>276.200104</z><t>2.762001</t></pt><pt><x>0.480214</x><y>0.295624</y><z>279.200012</z><t>2.792</t></pt><pt><x>0.478548</x><y>0.29019</y><z>282.400085</z><t>2.824001</t></pt><pt><x>0.472117</x><y>0.281429</y><z>287.400085</z><t>2.874001</t></pt><pt><x>0.4629</x><y>0.273228</y><z>288.899994</z><t>2.889</t></pt><pt><x>0.454855</x><y>0.267475</y><z>293.700012</z><t>2.937</t></pt><pt><x>0.450915</x><y>0.265296</y><z>297.000092</z><t>2.970001</t></pt><pt><x>0.452357</x><y>0.266478</y><z>303.30011</z><t>3.033001</t></pt><pt><x>0.456952</x><y>0.268994</y><z>306.600189</z><t>3.066002</t></pt><pt><x>0.462698</x><y>0.271786</y><z>310.5</z><t>3.105</t></pt><pt><x>0.468491</x><y>0.273698</y><z>313.700104</z><t>3.137001</t></pt><pt><x>0.472686</x><y>0.273481</y><z>317.000183</z><t>3.170002</t></pt><pt><x>0.474128</x><y>0.2711</y><z>320.200012</z><t>3.202</t></pt><pt><x>0.473317</x><y>0.26629</y><z>323.400085</z><t>3.234001</t></pt><pt><x>0.468818</x><y>0.258619</y><z>326.599915</z><t>3.265999</t></pt><pt><x>0.459177</x><y>0.249783</y><z>329.799988</z><t>3.298</t></pt><pt><x>0.444908</x><y>0.240565</y><z>333.000183</z><t>3.330002</t></pt><pt><x>0.428812</x><y>0.231547</y><z>336.200012</z><t>3.362</t></pt><pt><x>0.414001</x><y>0.224355</y><z>339.400085</z><t>3.394001</t></pt><pt><x>0.402203</x><y>0.220788</y><z>342.599915</z><t>3.425999</t></pt><pt><x>0.395916</x><y>0.220793</y><z>347.499817</z><t>3.474998</t></pt><pt><x>0.396958</x><y>0.22488</y><z>349.100098</z><t>3.491001</t></pt><pt><x>0.404304</x><y>0.233281</y><z>353.900085</z><t>3.539001</t></pt><pt><x>0.417089</x><y>0.244201</y><z>357.300232</z><t>3.573002</t></pt><pt><x>0.432787</x><y>0.255414</y><z>360.5</z><t>3.605</t></pt><pt><x>0.450306</x><y>0.266164</y><z>363.700104</z><t>3.637001</t></pt><pt><x>0.467561</x><y>0.276066</y><z>367.000183</z><t>3.670002</t></pt><pt><x>0.482574</x><y>0.283265</y><z>370.200012</z><t>3.702</t></pt><pt><x>0.49378</x><y>0.286947</y><z>373.400085</z><t>3.734001</t></pt><pt><x>0.499635</x><y>0.286057</y><z>376.699799</z><t>3.766998</t></pt><pt><x>0.499834</x><y>0.281879</y><z>379.899994</z><t>3.799</t></pt><pt><x>0.495725</x><y>0.275355</y><z>383.100098</z><t>3.831001</t></pt><pt><x>0.489335</x><y>0.268878</y><z>386.399811</z><t>3.863998</t></pt><pt><x>0.48229</x><y>0.263009</y><z>389.5</z><t>3.895</t></pt><pt><x>0.477876</x><y>0.259754</y><z>392.700195</z><t>3.927002</t></pt><pt><x>0.478274</x><y>0.260457</y><z>399.200104</z><t>3.992001</t></pt><pt><x>0.483361</x><y>0.262968</y><z>407.099915</z><t>4.070999</t></pt><pt><x>0.491255</x><y>0.266719</y><z>410.399994</z><t>4.104</t></pt><pt><x>0.499865</x><y>0.270938</y><z>413.600189</z><t>4.136002</t></pt><pt><x>0.506783</x><y>0.27343</y><z>416.799896</z><t>4.167999</t></pt><pt><x>0.508997</x><y>0.273246</y><z>420.000092</z><t>4.200001</t></pt><pt><x>0.504803</x><y>0.270227</y><z>426.399994</z><t>4.264</t></pt><pt><x>0.494368</x><y>0.264971</y><z>429.600098</z><t>4.296001</t></pt><pt><x>0.480507</x><y>0.258226</y><z>432.799896</z><t>4.327999</t></pt><pt><x>0.465572</x><y>0.250609</y><z>436.000092</z><t>4.360001</t></pt><pt><x>0.451107</x><y>0.242831</y><z>440.80011</z><t>4.408001</t></pt><pt><x>0.438001</x><y>0.233931</y><z>443.999908</z><t>4.439999</t></pt><pt><x>0.426576</x><y>0.223596</y><z>447.200012</z><t>4.472</t></pt><pt><x>0.418341</x><y>0.211109</y><z>450.400208</z><t>4.504002</t></pt><pt><x>0.413331</x><y>0.197413</y><z>453.599915</z><t>4.535999</t></pt><pt><x>0.409926</x><y>0.183593</y><z>456.80011</z><t>4.568001</t></pt><pt><x>0.408487</x><y>0.169777</y><z>459.999817</z><t>4.599998</t></pt><pt><x>0.409104</x><y>0.158001</y><z>463.200012</z><t>4.632</t></pt><pt><x>0.410391</x><y>0.149799</y><z>466.400085</z><t>4.664001</t></pt><pt><x>0.412971</x><y>0.146072</y><z>469.599915</z><t>4.695999</t></pt><pt><x>0.417036</x><y>0.147149</y><z>472.80011</z><t>4.728001</t></pt><pt><x>0.423736</x><y>0.154538</y><z>477.500183</z><t>4.775002</t></pt><pt><x>0.434048</x><y>0.168435</y><z>480.799896</z><t>4.807999</t></pt><pt><x>0.448041</x><y>0.1876</y><z>484</z><t>4.84</t></pt><pt><x>0.465073</x><y>0.211623</y><z>487.30011</z><t>4.873001</t></pt><pt><x>0.485318</x><y>0.239288</y><z>490.499908</z><t>4.904999</t></pt><pt><x>0.507639</x><y>0.268973</y><z>493.700012</z><t>4.937</t></pt><pt><x>0.531515</x><y>0.297824</y><z>496.900208</z><t>4.969002</t></pt><pt><x>0.554894</x><y>0.323854</y><z>500.099915</z><t>5.000999</t></pt><pt><x>0.575158</x><y>0.345687</y><z>503.30011</z><t>5.033001</t></pt><pt><x>0.590138</x><y>0.359054</y><z>506.399902</z><t>5.063999</t></pt><pt><x>0.597899</x><y>0.36054</y><z>509.700012</z><t>5.097</t></pt><pt><x>0.595797</x><y>0.348978</y><z>512.900208</z><t>5.129002</t></pt><pt><x>0.584151</x><y>0.326982</y><z>516.099915</z><t>5.160999</t></pt><pt><x>0.568078</x><y>0.299693</y><z>520.900024</z><t>5.209</t></pt><pt><x>0.551897</x><y>0.270388</y><z>522.499817</z><t>5.224998</t></pt><pt><x>0.536668</x><y>0.237784</y><z>527.299927</z><t>5.272999</t></pt><pt><x>0.519218</x><y>0.201608</y><z>530.5</z><t>5.305</t></pt><pt><x>0.498065</x><y>0.166022</y><z>533.700195</z><t>5.337002</t></pt><pt><x>0.474731</x><y>0.13713</y><z>536.899902</z><t>5.368999</t></pt><pt><x>0.45232</x><y>0.116845</y><z>540.100098</z><t>5.401001</t></pt><pt><x>0.433557</x><y>0.103536</y><z>543.299927</z><t>5.432999</t></pt><pt><x>0.420141</x><y>0.095593</y><z>546.5</z><t>5.465</t></pt><pt><x>0.414417</x><y>0.093934</y><z>549.700195</z><t>5.497002</t></pt><pt><x>0.416574</x><y>0.101241</y><z>556.200012</z><t>5.562</t></pt><pt><x>0.425488</x><y>0.115549</y><z>561.000122</z><t>5.610001</t></pt><pt><x>0.439859</x><y>0.136466</y><z>562.599915</z><t>5.625999</t></pt><pt><x>0.457912</x><y>0.162976</y><z>567.400024</z><t>5.674</t></pt><pt><x>0.477349</x><y>0.190422</y><z>570.500183</z><t>5.705002</t></pt><pt><x>0.495036</x><y>0.215822</y><z>573.799927</z><t>5.737999</t></pt><pt><x>0.50843</x><y>0.234787</y><z>577</z><t>5.77</t></pt><pt><x>0.516385</x><y>0.244551</y><z>580.200195</z><t>5.802002</t></pt><pt><x>0.521327</x><y>0.246433</y><z>583.400024</z><t>5.834</t></pt><pt><x>0.525809</x><y>0.245073</y><z>586.600098</z><t>5.866001</t></pt><pt><x>0.530392</x><y>0.245618</y><z>589.89978</z><t>5.898998</t></pt></stroke></drawing></tag></gml>"`;

exports[`parses a minimal GML document correctly 1`] = `"<gml spec=\\"1.0 (minimum)\\"><tag><drawing><stroke><pt><x>0</x><y>0</y></pt></stroke></drawing></tag></gml>"`;
